src/Pi.cpp

```diff
@@ -1107,6 +1107,21 @@ void GameLoop::Update(float deltaTime)
 	Pi::statSceneTris = 0;
 	Pi::statNumPatches = 0;
 
+	{
+		SDL_Window *w = Pi::renderer->GetSDLWindow();
+		if (w != NULL) {
+			static bool lastFocus = true;
+			uint32_t flags = SDL_GetWindowFlags(w);
+			bool hasFocus = (flags & SDL_WINDOW_INPUT_FOCUS) != 0;
+			if (hasFocus != lastFocus) {
+				lastFocus = hasFocus;
+				LuaEvent::Queue(hasFocus? "onFocusGained" : "onFocusLost");
```

Comment:
The proper place to handle this would be to listen for `SDL_WINDOWEVENT_FOCUS_GAINED` et. al. in `core/GuiApplication.cpp`, specifically the PollEvents function. That should dispatch to a virtual member method:

```cpp
virtual void OnWindowFocusChanged(bool newFocus) {}
```

That method would then be implemented by `Pi::App`, and an event dispatched to the UI event queue like so:

```cpp
void Pi::App::OnWindowFocusChanged(bool newFocus)
{
    if (!PiGui::GetEventQueue().IsValid())
        return;
    LuaEvent::Queue(PiGui::GetEventQueue(), newFocus ? "onFocusGained" : "onFocusLost");
}
```

Reply:
Thank you; implemented in b5e8818e7474e208ba8bd01a13dc823317f18737 (to be squashed)